README.md

```diff
@@ -124,22 +126,128 @@ add('X');
 
 ### Types
 
-sanctuary-def defines `Type` values for JavaScript's built-in types:
+Conceptually, a type is a set of values. One can think of a value of
+type `Type` as a function of type `Any -> Boolean` which tests values
+for membership in the set (though this is an oversimplification).
 
-  - `$.Array :: Type -> Type`
-  - `$.Boolean :: Type`
-  - `$.Date :: Type`
-  - `$.Error :: Type`
-  - `$.Function :: Type`
-  - `$.Null :: Type`
-  - `$.Number :: Type`
-  - `$.Object :: Type`
-  - `$.RegExp :: Type`
-  - `$.String :: Type`
-  - `$.Undefined :: Type`
+#### `Any`
 
-sanctuary-def also defines `$.Any`, the type of which every JavaScript value
-is a member.
+```haskell
+$.Any :: Type
+```
+
+Type comprising every JavaScript value.
+
+#### `Array`
+
+```haskell
+$.Array :: Type -> Type
+```
+
+Constructor for homogeneous Array types.
+
+#### `Boolean`
+
+```haskell
+$.Boolean :: Type
+```
+
+Type comprising `true` and `false` (and their object counterparts).
+
+#### `Date`
+
+```haskell
+$.Date :: Type
+```
+
+Type comprising every Date value.
+
+#### `Error`
+
+```haskell
+$.Error :: Type
+```
+
+Type comprising every Error value, including values of more specific
+constructors such as [`SyntaxError`][2] and [`TypeError`][3].
+
+#### `Function`
+
+```haskell
+$.Function :: Type
+```
+
+Type comprising every Function value.
+
+#### `Null`
+
+```haskell
+$.Null :: Type
+```
+
+Type whose sole member is `null`.
+
+#### `Number`
+
+```haskell
+$.Number :: Type
+```
+
+Type comprising every Number value (including `NaN` and Number objects).
+
+#### `Object`
+
+```haskell
+$.Object :: Type
+```
+
+Type comprising every "plain" Object value. Specifically, values created via:
+
+  - object literal syntax;
+  - `Object.create(null)`; or
```

Comment:
Good point. This shows how little I've used `Object.create`. Shall we simply replace "`Object.create(null)`" with "[`Object.create`][1]"?


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create